lib/SILOptimizer/Differentiation/VJPCloner.cpp

```diff
@@ -1009,16 +1009,269 @@ class VJPCloner::Implementation final
 
   void visitTryApplyInst(TryApplyInst *tai) {
```

Comment:
Completely optional, but it would be very nice if you could split the logic in this function into several helpers. Right now the function is quite big and it's a bit painful to dive into this w/o having full understanding of the context. You already have different pieces of logic wrapped into scopes `{/*...*/}`, so putting them into named helpers does not look too challenging. I'm sure we'll thank ourselves in future when we need to modify this :)

Feel free to ignore.

Reply:
Believe me or not, I tried to to do this multiple times :) The reality is that they are like 50% of same and 50% of different (between `apply` and `tryApply` lowering) and these differences are intermixed. There are also lots of values here and there so factoring out looked very artificial.